pkg/cmd/repo/clone/clone.go

```diff
@@ -142,9 +137,12 @@ func cloneRun(opts *CloneOptions) error {
 	}
 	canonicalCloneURL := ghrepo.FormatRemoteURL(canonicalRepo, protocol)
 
-	// If RepoHasWiki is true and repo opts has wiki then create a new clone URL
-	if canonicalRepo.RepoHasWiki() && wikipref {
-		canonicalCloneURL = strings.Replace(canonicalCloneURL, ".git", ".wiki.git", 1)
+	// If repo HasWikiEnabled and wantsWiki is true then create a new clone URL
+	if wantsWiki {
+		if !canonicalRepo.HasWikiEnabled {
+			return fmt.Errorf("The '%s' repository does not have a wiki or the wiki is not exported by the owner", ghrepo.FullName(canonicalRepo))
```

Comment:
I'm not sure what the “or the wiki is not exported by the owner” part refers to. Is it necessary to say that? I always thought that a repo can either have a wiki turned off or on; nothing in-between.

Reply:
![image](https://user-images.githubusercontent.com/6939749/100877695-26dfae00-34cf-11eb-982b-b7af65403b93.png)I got it from running this command on git. I'll update the error message to be `The '%s' repository does not have a wiki`.
And I'll add the Test case for `Test_RepoClone`.